### PR TITLE
Bug 1949153: Use centos/ruby-27-centos7 

### PR DESCRIPTION
**Pull request by dulek**
As the openshift/ruby-hello-world was updated to Ruby 2.7 a ton of tests
keep failing. This commit replaces references to ruby-25-centos7 with
ruby-27-centos7.

### CODE REVIEW

**Comment by mtnbikenc** [comment]
`'ipaddress` package issue should be resolved by https://github.com/openshift/openshift-ansible/pull/12326

**Comment by dulek** [comment]
/retest

Let's see if Jenkins jobs are fixed?

**Comment by dulek** [comment]
/retest

Okay, so what's failing now is still related to ruby app - looks like one test is wrongly looking for latest in the imagestream and others are hitting DockerHub pull limits. :/

**Comment by mtnbikenc** [comment]
The Ansible failure in e2e-gcp should be fixed by https://github.com/openshift/openshift-ansible/pull/12326/commits/cf969652514a375708473c2b38c25fcdd8af6746 when that PR merges.  I noticed that the e2e-gcp test is not exiting when Ansible fails and the job will just time out.  Must need some traps in that template.

**Comment by mtnbikenc** [comment]
https://github.com/openshift/openshift-ansible/pull/12326 is merged.
/retest

**Comment by mtnbikenc** [comment]
e2e-gcp passed. :tada:

**Comment by dulek** [comment]
@mtnbikenc: I created https://github.com/openshift/ruby-hello-world/pull/127 to fight the failures caused by DockerHub pull limits. 

cc @bparees

**Comment by dulek** [comment]
/retest

Those errors are different than previously. :/

**Comment by dulek** [comment]
@bparees, @mfojtik: Seems like there are new problems with those two openshift-jenkins jobs. Can we consider overriding them as this unblocks e2e job allowing to merge openshift-ansible/12316?

**Comment by dulek** [comment]
@bparees, @mfojtik: Well, seems like it only requires `/lgtm` and `/approve`. Would you consider reviewing it given this is unblocking at least e2e-gcp job?

**Comment by bparees** [comment]
/lgtm

it also needs a cherry pick approval.  i'm not sure how we're handling 3.11 merges in terms of cherry picks.  I assume we're good to merge whenever, but maybe @brenton or @vikaslaad can confirm.

**Comment by mtnbikenc** [comment]
Needs https://github.com/openshift/release/pull/18516 to fix cherry-pick-approved requirement.

**Comment by bparees** [comment]
> Needs openshift/release#18516 to fix cherry-pick-approved requirement.

since the label isn't supposed to be required anyway, i've just added it for now to bypass the requirement.

File: examples/sample-app/application-template-dockerbuild.json [hunk 1]
```diff
@@ -83,14 +83,14 @@
...
-        "name": "ruby-25-centos7"
+        "name": "ruby-27-centos7"
       },
       "spec": {
         "tags": [
           {
             "from": {
               "kind": "DockerImage",
-              "name": "docker.io/centos/ruby-25-centos7:latest"
+              "name": "docker.io/centos/ruby-27-centos7:latest"
```
Comment (xiuwang) [inline examples/sample-app/application-template-dockerbuild.json:93]: Due to docker hub pull limit, we may use quay.io image `quay.io/centos7/ruby-27-centos7`

File: examples/sample-app/application-template-pullspecbuild.json [hunk 2]
```diff
@@ -83,10 +83,10 @@
       "kind": "ImageStream",
       "apiVersion": "v1",
       "metadata": {
-        "name": "ruby-25-centos7"
+        "name": "ruby-27-centos7"
       },
       "spec": {
-        "dockerImageRepository": "centos/ruby-25-centos7"
+        "dockerImageRepository": "centos/ruby-27-centos7"
```
Comment (xiuwang) [inline examples/sample-app/application-template-pullspecbuild.json:89]: Ditto

**Comment by xiuwang** [comment]
How about using quay.io image quay.io/centos7/ruby-27-centos7 to replace ruby-25-centos?

**Pull request merged**